internal/cluster/transition.go

```diff
@@ -532,6 +534,8 @@ func (th *transitionHandler) InstallCluster(sw stateswitch.StateSwitch, args sta
 	// send metric and event that installation process has been started
 	params.metricApi.InstallationStarted(cluster.OpenshiftVersion, *cluster.ID, cluster.EmailDomain, strconv.FormatBool(swag.BoolValue(cluster.UserManagedNetworking)))
 	params.metricApi.ClusterHostInstallationCount(cluster.EmailDomain, len(cluster.Hosts), cluster.OpenshiftVersion)
+	captureMetricsForNetworkLatency(ctx, params.metricApi, cluster.OpenshiftVersion, cluster.Hosts)
```

Comment:
I don't think that InstallCluster is the right location to collect latency metrics.  What about metrics before installation?

Reply:
What do you mean before installation? 
Doing it once here reduces the amount of cycles required to process the connectivity report and also guarantees that the reporting information contains only the latest update. Correct me if I'm mistaken, but doing it at the place you suggested can lead to situations where the latency and packet loss information of a host can be reprocessed more than once if there are new hosts being added to the cluster, or even disabled/re-enabled, or new interfaces added, right?

Reply:
I think users expect to see the metrics before starting installation.  In addition, should these metrics captured once only, or when connectivity results arrive ?

Reply:
I'm not sure the metrics are targeted to be displayed to the users. If we want to display the current latency information, that could be shown in the UI, but at this point the only way to visualize the metrics will be in Grafana.
As for when the metrics should be captured, it's up to us to decide. I favor the option to capture them when the cluster is about to be installed so that we don't have polluted values, in which hosts that have high latency are not eventually added to the cluster for this or other reasons, or even glitches in the network that cause high latency during the discovery.  
If the general consensus is to capture any latency as it is reported to the service, then your suggestion is the right place to record the metrics. WDYT @avishayt @gamli75 ?

Reply:
The decision that was made by Ronnie & Avishay is to report the metric every time we receive the connectivity report and not only once before installation.